doc/nrf/glossary.rst

```diff
@@ -0,0 +1,447 @@
+
+Glossary
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Anycast addressing
+   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
+   This is a one-to-nearest association.
+
+Access Port Protection (APPROTECT)
+   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.
+
+API Result
+   Indicates success or failure of an API procedure.
+   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.
+
+Application
+   An implementation example that includes a variety of libraries to implement a specific use case.
+
+Application Programming Interface (API)
+   A language and message format used by an application program to communicate with an operating system, application, or other services.
+
+Board
+   In Zephyr and the |NCS|, a data structure for describing hardware on the board and its configuration files.
+
+Branch
+   A line of development composed of a sequence of commits.
+
+Carrier Wave (CW)
+   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.
+
+Cat-M1
+   LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.
+
+Cat-NB1
+   Narrowband Internet of Things (NB-IoT) User Equipment (UE) category with 200 kHz UE bandwidth and a single RX antenna, specified in 3GPP Release 13.
+
+Chain of Trust
+   A sequence of properties identifying the trustworthiness of each layer in the system, all the way back to a property in the system referred to as *root of trust* (RoT).
+   A secure system depends on building and maintaining a chain of trust through all the layers in the system.
+   Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.
+
+Clone
+   A local copy of a remote Git repository obtained using the command ``git clone``.
+
+Commissioning
+   In a thread mesh network, the process of authenticating and joining new devices to the network.
+
+Commit
+   A snapshot of the history of each project file, taken at a specific moment in time.
+   It is associated with a unique SHA and a message describing the edits it contains.
+
+Commit tag
+   A tag prepended to the first line of the commit message to ease filtering and identification of particular commit types.
+
+Contribution
+   A change to the codebase sent to a remote repository for inclusion.
+
+Development Kit (DK)
+   A hardware development platform used for application development.
+
+Device Firmware Update (DFU)
+   A mechanism for upgrading the firmware of a device.
+
+Devicetree
+   A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, and interrupt lines.
```

Comment:
I think data structure is appropriate; it is at least a tree, and spiritually it's a DAG since nodes can point to each other but cycles are disallowed.

Reply:
fixed